design-system/packages/options/src/index.tsx

```diff
@@ -149,7 +151,11 @@ const defaultComponents = {
   IndicatorSeparator: null,
 };
 
-type OptionsProps = NamedProps<{ label: string; value: string; isDisabled?: boolean }, boolean>;
+type OptionsProps = Props<{ label: string; value: string; isDisabled?: boolean }, boolean>;
+
+const SearchContext = createContext<{ shouldDisplaySearchControl: boolean }>({
```

Comment:
Creating context here to pass `shouldDisplaySearchControl` through to the control component, rather than leveraging `selectProps` from react-select. As the recommended method for defining the types for this negates our ability to have different selectProps for different usages of `react-select` in our monorepo. 
(cc: @mitchellhamilton)